### DISCUSSION TITLE
file extensions as a list?

### DISCUSSION THREAD

**Discussion by maasdorp** [localauthor/zk#22] [Q&A] [answered]
Is something like this possible?

(zk-file-extension '("md" "org"))

Or is it important to pick one file extension and stick to it?

**Comment by localauthor** [comment, marked as answer]
It’s not possible currently, no. It is necessary to use a single file extension for all files.

Is there a specific rationale or use case for allowing multiple file types? Seems like it would complicate things quite a bit.

**Reply by maasdorp** [reply]
Thanks for your reply and yes it makes sense.

I am migrating from org-roam to your system because I like your design principles. So I already have content in org and am also invested in org-mode for most of my writing projects.
However, if I started over I probably would go markdown -- it seems that markdown is the format that you use for your own notes (perhaps because you migrated from The Archive).

**Comment by localauthor** [comment]
Zk tries to be major-mode agnostic, so you should be able to continue using org-mode without problems. Just customize the header to your liking.

And actually the file extension doesn’t matter that much, as long as it’s consistent.

For example, my files have md extensions, but the buffers are in org-mode, and the header format doesn’t follow a standard format.

Let me know what issues you run into. And give zk-index a try!

**Reply by maasdorp** [reply]
Thank you for the extra detail. 

> For example, my files have md extensions, but the buffers are in org-mode, and the header format doesn’t follow a standard format.

Mind explaining the advantage of doing writing org-syntax in an md-file? Or is it a historical accident because you had md-files already?

Will let you know how it goes! Thanks.

**Comment by localauthor** [comment]
No advantage, purely historical. :)
